### PR TITLE
add toml++

### PR DESCRIPTION
**Pull request by yaozhenghangma**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/t/toml++/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package("toml++")
+
```
Comment (waruqi) [inline packages/t/toml++/xmake.lua:2]: please add `set_kind("library", {headeronly = true})`

File: packages/t/toml++/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    set_urls("https://github.com/marzer/tomlplusplus/archive/refs/tags/v$(version).zip")
+    add_versions("2.5.0", "887dfb7025d532a3485e1269ce5102d9e628ddce8dd055af1020c7b10ee14248")
+
+    on_load(function (package)
+        package:add("deps", "cmake")
+    end)
+
+    on_install(function (package)
+        local configs = {}
+        import("package.tools.cmake").install(package, configs)
```
Comment (waruqi) [inline packages/t/toml++/xmake.lua:15]: please remove unused configs

`import("package.tools.cmake").install(package)`

File: packages/t/toml++/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    set_homepage("https://marzer.github.io/tomlplusplus/")
+    set_description("toml++ is a header-only TOML config file parser and serializer for C++17 (and later!).")
+
+    set_urls("https://github.com/marzer/tomlplusplus/archive/refs/tags/v$(version).zip")
+    add_versions("2.5.0", "887dfb7025d532a3485e1269ce5102d9e628ddce8dd055af1020c7b10ee14248")
+
+    on_load(function (package)
+        package:add("deps", "cmake")
+    end)
```
Comment (waruqi) [inline packages/t/toml++/xmake.lua:11]: only need `add_deps("cmake")` and remove on_load

File: packages/t/toml++/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package("toml++")
+
+    set_homepage("https://marzer.github.io/tomlplusplus/")
+    set_description("toml++ is a header-only TOML config file parser and serializer for C++17 (and later!).")
+
+    set_urls("https://github.com/marzer/tomlplusplus/archive/refs/tags/v$(version).zip")
+    add_versions("2.5.0", "887dfb7025d532a3485e1269ce5102d9e628ddce8dd055af1020c7b10ee14248")
+
+    set_kind("library", {headeronly = true})
```
Comment (waruqi) [inline packages/t/toml++/xmake.lua:9]: please move set_kind to header

set_kind
set_homepage

**Comment by yaozhenghangma** [comment]
Thank you for teaching me these.

**Pull request merged**